middlewares/isLoggedIn.js

```diff
@@ -0,0 +1,36 @@
+const JWT = require('jsonwebtoken');
+
+exports.isLoggedin = async (req, res, next) => {
+	if (req.cookies?.accessToken) {
+		//access-token is available
+		try {
+			const verified = await JWT.verify(
+				req.cookies.accessToken,
+				process.env.JWT_ACCESS_SECRET
+			);
+			return next();
+		} catch (error) {
+			console.log('Access token invalid/expired');
+		}
+	}
+
+	console.log(
+		'Hello refresh-token checking as access token might be invalid/unavailable in cookies'
+	);
+	if (req.cookies?.refreshToken) {
```

Comment:
This condition should not be executed when the access token is present, right?
This should be else-if.
Correct me if I am wrong here.

If I am correct, then this log message on line 18 should be moved into this conditional check

Reply:
> This condition should not be executed when the access token is present, right? This should be else-if. Correct me if I am wrong here.
> 
> If I am correct, then this log message on line 18 should be moved into this conditional check

Actually assume a case where we have accessToken key, so the else-if part will never be executed. Now we have access token key, but suppose it has a invalid value(might be set by end-user in cookie). So now as access Token key exist but it's invalid, so we should move to refresh-token phase. But due to else-if we will not be able to do that. That's why i used only if instead of else-if. Just in case there is any error in the above idea, please clarify me for changes.

Reply:
Agreed!
Thanks. Looks good.